upstairs/src/lib.rs

```diff
@@ -5567,6 +5561,11 @@ impl Upstairs {
 
         Ok(notify_guest)
     }
+
+    async fn ds_repair_address(&self, client_id: u8, addr: SocketAddr) {
+        let mut ds = self.downstairs.lock().await;
+        ds.ds_repair.insert(client_id, addr);
```

Comment:
I take it there is no issue with replacement here?  Downstairs will come and go and the repair address could be different each time, we would need to handle that.  Maybe we want to consider removing it as part of the re-attachment process?

Reply:
yeah, I think I see what you're getting at here. I'm thinking that I should remove the previous address in `re_new`, does that sound right?

Reply:
sounds good, done in a97e6cd